README.md

```diff
@@ -136,6 +136,17 @@ Options:
                                 Output cannot be used in bash eval.
 ```
 
+## COMPATIBILITY
+
+Bash 4.x and higher is the main target.  You can also use bash 3.2 (mostly for macOS) with the `-G` option
+to `docopts` binary, and `source docopts.sh --auto -G` ([example](examples/legacy_bash/sshdiff_with_docopts.sh)),
+which both avoid using bash 4.x associative arrays.
+
+The `docopts.sh` helper supports the use of `set -u`, which [gives an error](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html#The-Set-Builtin) on undefined variables in your scripts.
```

Comment:
I didn't document `docopts.sh`, but I realize it could be a good start for the new API too. We may introduce a section for it. I mainly produced working examples but no documentation for it.